figure.cpp

```diff
@@ -65,3 +65,15 @@ bool Bishop::canMove(char rowFrom, char laneFrom,
     return (abs(rowTo - rowFrom) == abs(laneTo - laneFrom));
 }
 
+bool Queen::canMove(char rowFrom, char laneFrom,
+                     char rowTo, char laneTo) {
+    return ((abs(rowTo - rowFrom) == abs(laneTo - laneFrom)) ||
+            (abs(rowTo - rowFrom) && !(laneTo - laneFrom)) ||
+            (abs(laneTo - laneFrom) && !(rowTo - rowFrom)));
```

Comment:
please refactor this bit to be more efficient (see implementation of Rook::canMove)